dbt/macros/nearest_pin_neighbors.sql

```diff
@@ -46,27 +35,32 @@
                     from
                         (
                             select
-                                dp.pin10,
-                                dp.x_3435,
-                                dp.y_3435,
+                                sp.pin10,
+                                sp.x_3435,
+                                sp.y_3435,
                                 loc.year,
                                 loc.pin10 as neighbor_pin10,
                                 st_distance(
-                                    st_point(dp.x_3435, dp.y_3435), loc.point
+                                    st_point(sp.x_3435, sp.y_3435), loc.point
                                 ) as dist
-                            from distinct_pins as dp
+                            from source_pins as sp
                             inner join
                                 pin_locations as loc
                                 on st_contains(
-                                    st_buffer(
-                                        st_point(dp.x_3435, dp.y_3435), {{ radius_km }}
-                                    ),
-                                    loc.point
+                                    st_buffer(sp.point, {{ radius_km }}), loc.point
                                 )
-                                and dp.pin10 != loc.pin10
```

Comment:
This conditional actually made the results incorrect. Since the indexing below skips 1, but the same PIN match is excluded, we're actually getting the _second_ nearest neighbor for every PIN using this logic.

Reply:
**[Praise]** Oof, yeah... good catch!